tests/robot-tests/tests/libs/admin-common.robot

```diff
@@ -494,34 +539,49 @@ user changes methodology status to Approved
     user clicks button  Update status
 
 user gives analyst publication owner access
-    [Arguments]  ${ANALYST_EMAIL}  ${PUBLICATION_NAME}
-    user clicks link  Manage  xpath://td[text()="${ANALYST_EMAIL}"]/..
-    user waits until page does not contain loading spinner
-
-    # stale element exception if you don't wait until it's enabled    
-    user waits until button is enabled  Add publication access
-    user scrolls to element  css:[name="selectedPublicationId"]
-    
-    user waits until element is enabled  css:[name="selectedPublicationId"]
+    [Arguments]  ${PUBLICATION_NAME}  ${ANALYST_EMAIL}=ees-analyst1@education.gov.uk
+    user goes to manage user  ${ANALYST_EMAIL}
     user selects from list by label  css:[name="selectedPublicationId"]  ${PUBLICATION_NAME}
-
     user waits until element is enabled  css:[name="selectedPublicationRole"]
     user selects from list by label  css:[name="selectedPublicationRole"]  Owner
     user clicks button  Add publication access
-
-user removes publication owner access from analyst 
-    [Arguments]  ${PUBLICATION_NAME}
-    user waits until element is enabled  css:[name="selectedPublicationId"]
-    user scrolls to element  css:[name="selectedPublicationId"]
-    user clicks element  testid:remove-publication-role-${PUBLICATION_NAME}
     user waits until page does not contain loading spinner
 
-user gives analyst release access 
-    [Arguments]    ${RELEASE_NAME}  ${ROLE}
-    user waits until element is enabled  css:[name="selectedReleaseId"]
+user gives release access to analyst
+    [Arguments]    ${RELEASE_NAME}  ${ROLE}  ${ANALYST_EMAIL}=ees-analyst1@education.gov.uk
+    user goes to manage user  ${ANALYST_EMAIL}
     user scrolls to element  css:[name="selectedReleaseId"]
     user selects from list by label  css:[name="selectedReleaseId"]  ${RELEASE_NAME}
     user waits until element is enabled  css:[name="selectedReleaseRole"]
     user selects from list by label  css:[name="selectedReleaseRole"]  ${ROLE}
     user clicks button  Add release access
-    user waits until page does not contain loading spinner
\ No newline at end of file
+    user waits until page does not contain loading spinner
+
+user removes publication owner access from analyst 
+    [Arguments]  ${PUBLICATION_NAME}  ${ANALYST_EMAIL}=ees-analyst1@education.gov.uk
+    user goes to manage user  ${ANALYST_EMAIL}
+    user scrolls to element  css:[name="selectedPublicationId"]
+    # NOTE: The below wait is to prevent a transient failure that occurs on the UI test pipeline due to the DOM not being fully rendered which
+    # causes issues with getting the 'selectedPublicationId' selector (staleElementException)
```

Comment:
I feel and know your pain.

Reply:
Haha - can't take credit for this one - was in a pre-existing set of steps that I moved here to reuse :)